cubeb-core/src/lib.rs

```diff
@@ -3,4 +3,453 @@
 // This program is made available under an ISC-style license.  See the
 // accompanying file LICENSE for details.
 
+#[macro_use]
+extern crate bitflags;
+
 pub mod ffi;
+pub mod binding;
+mod error;
+mod util;
+
+use binding::Binding;
+pub use error::Error;
+use std::{marker, ptr, str};
+use util::opt_bytes;
+
+/// An enumeration of possible errors that can happen when working with cubeb.
+#[derive(PartialEq, Eq, Clone, Debug, Copy)]
+pub enum ErrorCode {
+    /// GenericError
+    Error,
+    /// Requested format is invalid
+    InvalidFormat,
+    /// Requested parameter is invalid
+    InvalidParameter,
+    /// Requested operation is not supported
+    NotSupported,
+    /// Requested device is unavailable
+    DeviceUnavailable
+}
+
+#[derive(PartialEq, Eq, Clone, Debug, Copy)]
+pub enum SampleFormat {
+    S16LE,
+    S16BE,
+    S16NE,
+    Float32LE,
+    Float32BE,
+    Float32NE
+}
+
+/// This maps to the underlying stream types on supported platforms, e.g. Android.
```

Comment:
FWIW this is obsolete and you can leave it out or make a note to remove it (https://github.com/kinetiknz/cubeb/issues/346)